libmuscle/cpp/src/libmuscle/tests/test_snapshot_manager.cpp

```diff
@@ -0,0 +1,230 @@
+// Inject mocks
+#define LIBMUSCLE_MOCK_COMMUNICATOR <mocks/mock_communicator.hpp>
+#define LIBMUSCLE_MOCK_LOGGER <mocks/mock_logger.hpp>
+#define LIBMUSCLE_MOCK_MMP_CLIENT <mocks/mock_mmp_client.hpp>
+#define LIBMUSCLE_MOCK_PROFILER <mocks/mock_profiler.hpp>
+
+// into the real implementation,
+#include <ymmsl/ymmsl.hpp>
+
+#include <libmuscle/close_port.cpp>
+#include <libmuscle/data.cpp>
+#include <libmuscle/instance.cpp>
+#include <libmuscle/logging.cpp>
+#include <libmuscle/mcp/data_pack.cpp>
+#include <libmuscle/message.cpp>
+#include <libmuscle/port.cpp>
+#include <libmuscle/settings_manager.cpp>
+#include <libmuscle/snapshot_manager.cpp>
+#include <libmuscle/timestamp.cpp>
+
+// then add mock implementations as needed.
+#include <mocks/mock_communicator.cpp>
+#include <mocks/mock_logger.cpp>
+#include <mocks/mock_mmp_client.cpp>
+#include <mocks/mock_profiler.cpp>
+
+// Test code dependencies
+#include <cstdio>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <typeinfo>
+
+#include <gtest/gtest.h>
+#include <libmuscle/snapshot_manager.hpp>
+
+// Note: using POSIX for filesystem calls
+// Could be upgraded to std::filesystem when targeting C++17 or later
+#include <unistd.h>
+#include <ftw.h>
+
+using libmuscle::impl::Data;
+using libmuscle::impl::Message;
+using libmuscle::impl::MockCommunicator;
+using libmuscle::impl::MockLogger;
+using libmuscle::impl::MockMMPClient;
+using libmuscle::impl::MockProfiler;
+using libmuscle::impl::Optional;
+using libmuscle::impl::Snapshot;
+using libmuscle::impl::SnapshotMetadata;
+using libmuscle::impl::SnapshotManager;
+using ymmsl::Reference;
+
+int main(int argc, char *argv[]) {
+    ::testing::InitGoogleTest(&argc, argv);
+    return RUN_ALL_TESTS();
+}
+
+/* Mocks have internal state, which needs to be reset before each test. This
+ * means that the tests are not reentrant, and cannot be run in parallel.
+ * It's all fast enough, so that's not a problem.
+ */
+void reset_mocks() {
+    MockCommunicator::reset();
+    MockMMPClient::reset();
+}
+
+MockLogger & mock_logger() {
+    static MockLogger logger;
+    return logger;
+}
+
+MockProfiler & mock_profiler() {
+    static MockProfiler profiler;
+    return profiler;
+}
+
+class libmuscle_snapshot_manager : public ::testing::Test {
+    protected:
+        void SetUp() override {
+            auto tmpnam = std::tmpnam(nullptr);
+            temp_dir_ = std::string(tmpnam);
+            if (mkdir(tmpnam, 0777) < 0)
```

Comment:
I figured it would no be too big of an issue for unit tests, but updated regardless 😄

Reply:
I'm extra sensitive to these things because temporary files and directories are a big source of security vulnerabilities. That's not super-relevant in this context, but I still feel better doing it right.